src/map/packet_guard.cpp

```diff
@@ -0,0 +1,57 @@
+﻿
+#include "packet_guard.h"
+
+#include "entities/charentity.h"
+
+#include <unordered_map> // Lookup
+#include <unordered_set> // Capture
+
+// #define PACKETGUARD_CAP_ENABLED 1
+
+namespace PacketGuard
+{
+#ifdef PACKETGUARD_CAP_ENABLED
+std::unordered_map<CHAR_SUBSTATE, std::unordered_set<uint16>> regular_client_packets;
+#endif
+
+std::unordered_map<CHAR_SUBSTATE, std::unordered_map<uint16, bool>> allowlist;
+
+void Init()
+{
+    // Allow all non-substate packets
+    for (uint16 i = 0; i < 512; ++i)
+    {
+        allowlist[SUBSTATE_NONE][i] = true;
+    }
+
+    // In Cutscene
+    allowlist[SUBSTATE_IN_CS][0x015] = true; // Player Sync
+    allowlist[SUBSTATE_IN_CS][0x016] = true; // Entity Information Request
+    allowlist[SUBSTATE_IN_CS][0x01A] = true; // Player Action
+    allowlist[SUBSTATE_IN_CS][0x03A] = true; // Sort Inventory
+    allowlist[SUBSTATE_IN_CS][0x05B] = true; // Event Update (Completion or Update)
+    allowlist[SUBSTATE_IN_CS][0x05C] = true; // Event Update (Update Player Position)
+    allowlist[SUBSTATE_IN_CS][0x0B5] = true; // Chat Message
```

Comment:
```
0x0B6 // Tell
```

I'm _fairly_ certain party commands are allowed during CS too. These would be [0x06E through 0x078](https://github.com/Windower/Lua/blob/dev/addons/libs/packets/fields.lua#L740-L791).

But I do play on retail with `autoinvite` and `autojoin`, so my memory may be influenced through packet injections as opposed to vanilla commands. Someone may want to check with a vanilla client on retail if you can `/pcmd add`, `/join`, `/pcmd leader`, and `/pcmd leave` during a CS event (Eternal Flame in Western Adoulin counts as a CS).

Reply:
I did try `/pcmd add` I think, but I will double check all of these that you mentioned, thanks!

Reply:
Confirmed: `/pcmd leave` is `You cannot use that command at this time.`

Reply:
Same with `/join` and all the others